controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -218,20 +216,21 @@ function fetch($request, $actionArgs = array()) {
 					'assignedToSubmissionStage' => $query->getStageId(),
 				),
 				// Include the full name and role in this submission in the item title
-				'setItemTitleCallback' => function($user, $userProps) use ($assignments) {
+				'setItemTitleCallback' => function($user, $userProps) use ($stageAssignmentDao, $query) {
 					$title = $user->getFullName();
-					foreach ($assignments as $assignment) {
-						if ($assignment->getUserId() === $user->getId()) {
-							foreach ($userProps['groups'] as $userGroup) {
-								if ($userGroup['id'] === (int) $assignment->getUserGroupId() && isset($userGroup['name'][AppLocale::getLocale()])) {
-									$title =  __('submission.query.participantTitle', array(
-										'fullName' => $user->getFullName(),
-										'userGroup' => $userGroup['name'][AppLocale::getLocale()],
-									));
-								}
+					$usersAssignments = $stageAssignmentDao->getBySubmissionAndStageId($query->getAssocId(), $query->getStageId(), null, $user->getId())->toArray();
+					$userRoles = '';
+					foreach ($usersAssignments as $assignment) {
+						foreach ($userProps['groups'] as $userGroup) {
+							if ($userGroup['id'] === (int) $assignment->getUserGroupId() && isset($userGroup['name'][AppLocale::getLocale()])) {
+								$userRoles .= $userGroup['name'][AppLocale::getLocale()] . ", ";
```

Comment:
We now have a locale string, `common.listSeparator`, to use for cases like this.

Also, no big deal, but I like using arrays and `join` to do list separation:

```php
$userRoles = ['Editor', 'Author'];
echo join(__('common.listSeparator'), $userRoles);
```

If you stick with the string concatenation, that's fine. I can't say one way is "better", but you'll need to adjust the `rtrim` method below too for cases where it has a different translation.

Reply:
much better way, thanks!